test/core/filters/filter_fuzzer.cc

```diff
@@ -0,0 +1,571 @@
+      : engines_(engines) {}
+  void Orphan() override {}
+  AuthorizationEngines engines() override { return engines_; }
+
+ private:
+  AuthorizationEngines engines_;
+};
+
+struct GlobalObjects {
+  ResourceQuotaRefPtr resource_quota = MakeResourceQuota("test");
+  grpc_transport transport{&kFakeTransportVTable};
+  RefCountedPtr<FakeChannelSecurityConnector> channel_security_connector{
+      MakeRefCounted<FakeChannelSecurityConnector>()};
+
+  void Perform(const filter_fuzzer::GlobalObjectAction& action) {
+    switch (action.type_case()) {
+      case filter_fuzzer::GlobalObjectAction::TYPE_NOT_SET:
+        break;
+      case filter_fuzzer::GlobalObjectAction::kSetResourceQuota:
+        resource_quota->memory_quota()->SetSize(action.set_resource_quota());
+        break;
+      case filter_fuzzer::GlobalObjectAction::kFinishCheckCallHost:
+        channel_security_connector->FinishCheckCallHost(
+            action.finish_check_call_host().qry(),
+            absl::Status(
+                absl::StatusCode(action.finish_check_call_host().status()),
+                action.finish_check_call_host().message()));
+        break;
+    }
+  }
+};
+
+struct Filter {
+  absl::string_view name;
+  absl::StatusOr<std::unique_ptr<ChannelFilter>> (*create)(
+      ChannelArgs channel_args, ChannelFilter::Args filter_args);
+
+  template <typename T>
+  static Filter* Make(absl::string_view name) {
+    return new Filter{
+        name,
+        [](ChannelArgs channel_args, ChannelFilter::Args filter_args)
+            -> absl::StatusOr<std::unique_ptr<ChannelFilter>> {
+          auto r = T::Create(channel_args, filter_args);
+          if (!r.ok()) return r.status();
+          return std::unique_ptr<ChannelFilter>(new T(std::move(*r)));
+        }};
+  }
+};
+
+RefCountedPtr<AuthorizationEngine> LoadAuthorizationEngine(
+    const filter_fuzzer::AuthorizationEngine& engine) {
+  switch (engine.engine_case()) {
+    case filter_fuzzer::AuthorizationEngine::kAlwaysAllow:
+      return MakeRefCounted<ConstAuthorizationEngine>(
+          AuthorizationEngine::Decision{
+              AuthorizationEngine::Decision::Type::kAllow,
+              engine.always_allow()});
+    case filter_fuzzer::AuthorizationEngine::kAlwaysDeny:
+      return MakeRefCounted<ConstAuthorizationEngine>(
+          AuthorizationEngine::Decision{
+              AuthorizationEngine::Decision::Type::kDeny,
+              engine.always_deny()});
+    case filter_fuzzer::AuthorizationEngine::ENGINE_NOT_SET:
+      break;
+  }
+  return MakeRefCounted<ConstAuthorizationEngine>(AuthorizationEngine::Decision{
+      AuthorizationEngine::Decision::Type::kAllow, engine.always_allow()});
+}
+
+template <typename FuzzerChannelArgs>
+ChannelArgs LoadChannelArgs(const FuzzerChannelArgs& fuzz_args,
+                            GlobalObjects* globals) {
+  ChannelArgs args = ChannelArgs().SetObject(ResourceQuota::Default());
```

Comment:
Can there be a case where a Filter needs some other specific type of object besides resource quota (e.g Transport) to be always present in the channel args ?

Reply:
yes-ish... 

if it's something we always guarantee to be present we should add it here,
if it's something we don't, then those filters should fail to instantiate